src/amo/reducers/landing.js

```diff
@@ -0,0 +1,36 @@
+import {
+  LANDING_GET,
+  LANDING_LOADED,
+  LANDING_FAILED,
+} from 'core/constants';
+
+
+export const initialState = {
+  featured: { count: 0, results: [] },
+  highlyRated: { count: 0, results: [] },
+  loading: false,
+  popular: { count: 0, results: [] },
+};
+
+export default function landing(state = initialState, action) {
+  const { payload } = action;
+  let newState;
+  switch (action.type) {
+    case LANDING_GET:
+      return { ...state, ...payload, loading: true };
```

Comment:
I don't think there should be a default value set but I've set one to `null` so the data structure is clearer. And I switched to `addonType: payload.addonType`; you're right the explicitness is better 👍